tidal/content/contents/code/tidal.js

```diff
@@ -129,76 +107,97 @@ var TidalResolver = Tomahawk.extend( TomahawkResolver, {
-                +query+"&token="+that.api_token+"&countryCode="+that._countryCode,
-            function (request) {
-                Tomahawk.log(request.responseText);
-                var resp = JSON.parse(request.responseText);
-                var data = {
-                    qid: qid,
-                    artists: resp.items.map(that._convertArtist),
-                }
-                Tomahawk.addArtistResults(data);
+            Tomahawk.addTrackResults(data);
+        });
+
+        // TODO: Return these too!
+
+        Tomahawk.ajax(this.api_location + "search/albums",
+            {
+                data: params
             }
-        );
+        ).then( function (response) {
+            var data = {
+                qid: qid,
+                albums: response.items.map(that._convertAlbum),
+            }
+            Tomahawk.addAlbumResults(data);
+        });
+
+        Tomahawk.ajax(this.api_location + "search/artists",
+            {
+                data: params
+            }
+        ).then( function (response) {
+            var data = {
+                qid: qid,
+                artists: response.items.map(that._convertArtist),
+            }
+            Tomahawk.addArtistResults(data);
+        });
     },
 
     resolve: function (qid, artist, album, title) {
+        Tomahawk.log('Called resolve()');
+        if (!this.logged_in) this._login(this.resolve, [qid, artist, album, title], this);
+
         var that = this;
-        if (!this._countryCode) return;
-        artist = encodeURIComponent(artist.replace('-',' '));
-        album = encodeURIComponent(album.replace('-',' '));
-        title = encodeURIComponent(title.replace('-',' '));
-
-        xhrRequest(that.api_location + "search/tracks","GET", "query=" 
-                +artist+"+"+album+"+"+title+"&token="+that.api_token+"&countryCode="+that._countryCode,
-            function (request) {
-                var resp = JSON.parse(request.responseText);
-                var data = {
-                    qid: qid,
-                    results: [],
-                }
-                if (resp.items.length != 0)
-                {
-                    data.results.push(that._convertTrack(resp.items[0]));
-                }
-                Tomahawk.addTrackResults(data);
+
+        var query = [
+            artist.replace('-', ' '),
+            album.replace('-', ' '),
+            title.replace('-', ' ')
+        ].join(' ');
+
+        query = query.replace(/ +/, ' ');
+
+        Tomahawk.log(query);
+
+        return Tomahawk.ajax(that.api_location + "search/tracks", {
+            data: {
+                query: query,
+                token: this.api_token,
+                countryCode: this._countryCode
             }
-        );
+        }).then(function (response) {
+            var data = {
+                qid: qid,
+                results: response.items.map(that._convertTrack),
+            }
+            Tomahawk.addTrackResults(data);
```

Comment:
I tried return response.items.map(that._convertTrack); But nothing seemed to happen.